tests/jballerina-unit-test/src/test/resources/test-src/object/ObjectProject/src/init/object-initializer.bal

```diff
@@ -241,3 +241,176 @@ function testCheckPanicInObjectInitArg() returns error {
 function testObjectInitPanic() returns error|PanicReceiver {
     return trap new PanicReceiver("Mr. Panic");
 }
+
+type Student1 object {
+     public int marks = 75;
+
+     public function __init() {
+         string grade = "B";
+     }
+
+     public function getMarks() returns int {
+         self.__init();
+         return self.marks;
+     }
+ };
+
+function testInitInvocationInsideObject() returns ([int, int]) {
+    Student1 student = new;
+    int marksBeforeChange = student.getMarks();
+    student.marks = 80;
+    int marksAfterChange = student.getMarks();
+    return [marksBeforeChange, marksAfterChange];
+}
+
+type Student2 object {
+    public int marks = 75;
+    public string grade;
+
+    public function __init(string grade) {
+        self.grade = grade;
+    }
+
+    public function getMarks() returns int {
+        self.__init("B+");
+        return self.marks;
+    }
+};
+
+function testInitInvocationInsideObjectWithArgs() returns ([int, int]) {
+    Student2 student = new("B");
+    int marksBeforeChange = student.getMarks();
+    student.marks = 82;
+    int marksAfterChange = student.getMarks();
+    return [marksBeforeChange, marksAfterChange];
+}
+
+type Student3 object {
+    public string name;
+    public int marks = 90;
+
+    public function __init(int id) returns error? {
+        self.name = check getName(id);
+    }
+
+    public function getMarks() returns int {
+        var v = self.__init(0);
+        return self.marks;
+    }
+};
+
+function getName(int id) returns string|error {
+    if (id == -1) {
+        return error("failed to return a name");
+    } else {
+        return "Smith";
+    }
+}
+
+function testInitInvocationWithReturn1() returns ([Student3|error, int, int]) {
+    int marksBeforeChange = 0;
+    int marksAfterChange = 0;
+
+    Student3|error student = new(-1);
+    if (student is error) {
+        return [student, marksBeforeChange, marksAfterChange];
+    }
+
+    Student3 s = <Student3> student;
+    marksBeforeChange = s.getMarks();
+    s.marks = 95;
+    marksAfterChange = s.getMarks();
+    return [s, marksBeforeChange, marksAfterChange];
```

Comment:
These lines never execute. 
The best way to model this test is, have a single testInitInvocationWithReturn method by taking an argument. Then use it in both testInitInvocationWithReturn1 and  testInitInvocationWithReturn2 test cases. 
Also, shall we have a better name for this function, which explains the test scenario? like. `testObjInitWithCheck`

Reply:
Done